author.tf

```diff
@@ -449,6 +500,72 @@ module "publisher" {
   EOF
 }
 
+module "survey-register" {
+  source                 = "github.com/ONSdigital/eq-ecs-deploy?ref=v4.1"
+  env                    = "${var.env}"
+  aws_account_id         = "${var.aws_account_id}"
+  aws_assume_role_arn    = "${var.aws_assume_role_arn}"
+  dns_zone_name          = "${var.dns_zone_name}"
+  ecs_cluster_name       = "${module.author-eq-ecs.ecs_cluster_name}"
+  vpc_id                 = "${module.author-vpc.vpc_id}"
+  aws_alb_arn            = "${module.author-eq-ecs.aws_external_alb_arn}"
+  aws_alb_listener_arn   = "${module.author-eq-ecs.aws_external_alb_listener_arn}"
+  service_name           = "survey-register"
+  listener_rule_priority = 600
+  docker_registry        = "${var.author_registry}"
+  container_name         = "eq-survey-register"
+  container_port         = 8080
+  container_tag          = "switching-register-to-node"
+  healthcheck_path       = "/status"
+  application_min_tasks  = "${var.survey_register_min_tasks}"
+  slack_alert_sns_arn    = "${module.eq-alerting.slack_alert_sns_arn}"
+  ecs_subnet_ids         = "${module.author-eq-ecs.ecs_subnet_ids}"
+  ecs_alb_security_group = ["${module.author-eq-ecs.ecs_alb_security_group}"]
+  launch_type            = "FARGATE"
+
+  container_environment_variables = <<EOF
+      {
+        "name": "DYNAMO_SURVEY_REGISTRY_TABLE_NAME",
+        "value": "${module.author-dynamodb.survey_registry_table_name}"
+      },
+      {
+        "name": "SURVEY_REGISTER_URL",
+        "value": "${module.survey-register.service_address}"
+      },
+      {
+        "name": "GO_QUICK_LAUNCHER_URL",
+        "value": "${module.author-survey-launcher.service_address}/quick-launch?url="
+      },
+      {
+        "name": "PUBLISHER_URL",
+        "value": "${module.publisher.service_address}/publish/"
+      }
+  EOF
+
+  task_has_iam_policy = true
+
+  task_iam_policy_json = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+      {
+          "Sid": "",
+          "Effect": "Allow",
+          "Action": [
+              "dynamodb:Scan",
+              "dynamodb:DescribeTable",
+              "dynamodb:PutItem",
+              "dynamodb:UpdateItem",
```

Comment:
The registry should be an immutable store of schemas. So when we change the registry to push new versions instead of overwriting the exiting one, we should probably remove the UpdateItem permission.